Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/NoHardCodedPathsAnalyzer.cs

```diff
@@ -45,14 +46,20 @@ private void Analyze(SyntaxNodeAnalysisContext context)
 				return;
 			}
 
-			//if the character of the string do not match either of the characters : for windows and / for linux; no need to run regex, simply return.
+			// If the character of the string do not match either of the characters : for windows and / for linux; no need to run regex, simply return.
 			if (!pathValue[1].Equals(':') && !pathValue[0].Equals('/'))
 			{
 				return;
 			}
 
+			// Limit to first 1000 characters, to prevent long analysis times.
+			if (pathValue.Length > MaxStringLength)
```

Comment:
I'd be tempted to use MAX_PATH, even though modern systems allow for more than that (with an opt-in, I believe?), on the grounds that a very long hardcoded path seems exceedingly unlikely.

Reply:
The Analyzer expects the path at the beginning of the string, but could also be in the middle...

Reply:
For now, left the regex such that it checks the beginning only. If the need arises we can always rework later on.